sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -19,14 +19,16 @@ export interface LongRunningOperation<T> {
 
 // @public
 export class LroEngine<TResult, TState extends PollOperationState<TResult>> extends Poller<TState, TResult> {
-    constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions);
+    constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions<TResult, TState>);
     delay(): Promise<void>;
 }
 
 // @public
-export interface LroEngineOptions {
+export interface LroEngineOptions<TResult, TState> {
     intervalInMs?: number;
     lroResourceLocationConfig?: LroResourceLocationConfig;
+    processResult?: (result: unknown, state: TState) => TResult;
+    processState?: (state: TState, lastResponse: RawResponse) => void;
```

Comment:
I feel like this should be called `updateState` or something similar, since unlike the above function, it's not a map over state that returns a new state, but a method that modifies the existing state in place.

Reply:
I agree, I updated the name for `processState` to be `updateState`.